discovery-provider/src/queries/notifications.py

```diff
@@ -46,8 +57,124 @@ def get_owner_id(session, entity_type, entity_id):
     else:
         return None
 
+def get_cosign_remix_notifications(session, max_block_number, remix_tracks):
+    """
+    Get the notifications for remix tracks that are reposted/favorited by the parent remix author
+
+    Args:
+        session: (DB)
+        max_block_number: (int)
+        remix_tracks: (Array<{ }>)
+            'user_id'
+            'item_id'
+            const.notification_blocknumber
+            const.notification_timestamp
+            'item_owner_id'
+    
+    Returns:
+        Array of cosign notifications
+
+    """
+    if len(remix_tracks) == 0:
+        return []
+
+    remix_notifications = []
+    remix_track_ids = [r['item_id'] for r in remix_tracks]
+
+    # Query for all the parent tracks of the remix tracks 
+    tracks_subquery = (
+        session.query(Track)
+            .filter(
+                Track.is_unlisted == False,
+                Track.is_delete == False,
+                Track.blocknumber <= max_block_number
+            )
+            .distinct(Track.track_id)
+            .order_by(desc(Track.track_id),desc(Track.blocknumber))
+            .subquery()
+    )
+
+    remix_track = (
+        session.query(
+            Remix.child_track_id,
+            Remix.parent_track_id,
+            tracks_subquery.c.owner_id
+        )
+        .outerjoin(
+            tracks_subquery,
+            Remix.parent_track_id == tracks_subquery.c.track_id
+        )
+        .filter(
+            Remix.child_track_id.in_(remix_track_ids)
+        )
+        .all()
+    )
+    # Mapping of parent track users to child track to parent track
+    track_parents = {}
+    for track_parent in remix_track:
+        [remix_track_id, remix_parent_id, remix_parent_user_id] = track_parent 
```

Comment:
I believe that when you query `session.query(Remix.child_track_id, Remix.parent_track_id, tracks_subquery.c.owner_id )` the response of the `query.all()` will return the query entities in the same order. Can't find the docs that say it's guaranteed, but it should map the sql responses in same order right? It worked for me when I tested locally at least and I don't see why it would return a different order.